force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,17 +133,28 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
-
+            String scientistGroup = '';
         // Loop through the list of scientists' names
+        for (String scientists : scientistNames) {
+            if (scientists.contains('g') || scientists.contains('G')) {
+                scientistGroup += scientists + ',';
+            }
+        }
+        String scientistGroupTrimmed = scientistGroup.removeEnd(',');
+        return scientistGroupTrimmed;
+    }
```

Comment:
Great work. Something I like to do, rather than checking for upper and lower case is to just turn the the whole thing into one care or another. 

```
 public static String findTheScientist() {
        // The list of scientists' names
        List<String> scientistNames = new List<String>{
                'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'
        };
        String result = '';
        for (String name : scientistNames) {
            if (name.toLowerCase().contains('g')) {
                result += name + ',';
            }
        }
        return result.removeEnd(',');
    }
 ```